csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -317,12 +305,6 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        ~DisposableNamedOnnxValue()
```

Comment:
Does removal of this destructor mean that the caller from C# would have to explicitly call dispose?

Reply:
There are no destructors in C#. These are called finalizers. They are added bc of Microsoft Documentation is really unclear what they are. 
They are generally useless and misleading especially for disposable objects. Such objects need release native critical resources, they need to be disposed quickly just as the destructor would do in C++.
Almost all of the C# classes in Onnxruntime expose native resources and they are all disposable.
Typically, people would use try/finally OR using() keyword that causes the compiler to generate try/finally. 
Unfortunately, this may lead to many levels nested using things and unmanageable functions.
The solution for that is write smaller functions. Also we have a helper, Disposable List where one can add disposable objects and have one using() clause that would dispose of that list and it in turn would dispose all contained objects.
If you wish more explanation it may be done over a call.

Reply:
Yes, that's true. I forgot that for managed languages there are no destructors :P. Ignore my comment. 

---
In reply to: [462676366](https://github.com/microsoft/onnxruntime/pull/4646#discussion_r462676366) [](ancestors = 462676366)